### PR TITLE
Implement Class Relantionships 

### PR DESCRIPTION
**Pull request by Nicolaswg**
# Main Point 🥇 

🦅 
> In this new PR, we implement  Class relationships. Specifically the association's relation between classes.

## Project requirements 📄 
- Created a `has-many/belongs-to` relationship between Classroom and Student
- Created a `many-to-many` relationship between Person and the Book using Rental as intermediate 

## Set up 💻 
- check [Understanding OOP relationships](https://shouts.dev/understanding-oop-concepts-association-composition-aggregation)
- Clone the PR repo in your local machine  `git clone  https://github.com/Nicolaswg/school-library-ruby.git `

>### Thanks for the review, any optional suggestions will be received

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative './rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (bimbolabuari) [inline book.rb]: Getters and Setters for `rentals` should be present, Kindly add the `rentals` to attr_accessor. 👍

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,16 @@
 require_relative './nameable'
+require_relative './rental'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (bimbolabuari) [inline person.rb]: Getters and Setters for `rentals` should be present, Kindly add the `rentals` to attr_accessor.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :book, :person
+  attr_accessor :date
```
Comment (bimbolabuari) [inline rental.rb]: Getters and Setters for `@book` and `@person` instance variables should be present. Kindly add them to the attr_accessor. 🙂

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(date, book, person)
+    @date = date
+
+    @book = book
+    book.retals << self
+
+    @person = person
+    person.retals << self
```
Comment (bimbolabuari) [inline rental.rb]: I think you have some typo error with the `rentals`, Please fix this 👍

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @Nicolaswg, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Set up associations between classes.
- Correct Gitflow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 4 above)
Comment (Nicolaswg) [reply rental.rb]: thanks for that I didn't notice the typo error 😄

**Review by bimbolabuari — APPROVED** [review]
Hi @Nicolaswg, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**